src/calldata/evm.ts

```diff
@@ -0,0 +1,366 @@
+import { Buffer } from 'buffer/';
+import { keccak256 } from 'js-sha3';
+import { encode } from 'rlp';
+
+/**
+* Look through an ABI definition to see if there is a function 
+* that matches the signature provided.
+* @param sig    a 0x-prefixed hex string containing 4 bytes of info
+* @param abi    a Solidity JSON ABI structure
+*               (https://docs.ethers.io/v5/api/utils/abi/formats/#abi-formats--solidity)
+* @returns      Buffer containing RLP-serialized array of 
+*               calldata info to pass to signing request
+*/ 
+export const parseSolidityJSONABI = function(sig: string, abi: any[]): Buffer {
+  sig = coerceSig(sig);
+  let match = null;
+  abi.forEach((item) => {
+    if (item.type === 'function') {
+      const def = parseDef(item);
+      if (def) {
+        // If this matches the function selector (sig) we can return it
+        if (def && getFuncSig(def.canonicalName) === sig) {
+          match = def.def;
+        }
+      }
+    }
+  })
+  if (match) {
+    return Buffer.from(encode(match));
+  }
+  return null;
+}
+
+/**
+* Convert a canonical name into an ABI definition that can be included
+* with calldata to a general signing request. Parameter names will be 
+* encoded in order that they are discovered (e.g. "1", "2", "2.1", "3")
+* @param sig    a 0x-prefixed hex string containing 4 bytes of info
+* @param name   canonical name of the function
+* @returns      Buffer containing RLP-serialized array of 
+*               calldata info to pass to signing request
+*/
+export const parseCanonicalName = function(sig: string, name: string): Buffer {
+  sig = coerceSig(sig);
+  if (sig !== getFuncSig(name)) {
+    throw new Error('Name does not match provided sig.')
+  }
+  const def = [];
+  // Get the function name
+  const paramStart = name.indexOf('(');
+  if (paramStart < 0) {
+    throw new Error(BAD_CANONICAL_ERR);
+  }
+  def.push(name.slice(0, paramStart));
+  name = name.slice(paramStart + 1)
+  let paramDef = [];
+  while (name.length > 1) { // scan until the terminating ')'
+    const typeStr = popTypeStrFromCanonical(name);
+    paramDef = paramDef.concat(parseTypeStr(typeStr));
+    name = name.slice(typeStr.length + 1)
+  }
+  const parsedParamDef = parseParamDef(paramDef);
+  return Buffer.from(encode(def.concat(parsedParamDef)));
+}
+
+// Convert a canonical name to a function selector (a.k.a. "sig")
+function getFuncSig(canonicalName: string): string {
+  return `0x${keccak256(canonicalName).slice(0, 8)}`
+}
+
+// Ensure the sig is properly formatted
+function coerceSig(sig: string): string {
+  if (typeof sig !== 'string' || (sig.length !== 10 && sig.length !== 8)) {
+    throw new Error('`sig` must be a hex string with 4 bytes of data.')
+  }
+  if (sig.length === 8) {
+    sig = `0x${sig}`;
+  }
+  return sig;
+}
+
+// Take the next type from a canonical definition string.
```

Comment:
If we use the `/** */` format for comments, we get better intellisense comments that pop up when hovering or using these functions.

Reply:
[756bc00](https://github.com/GridPlus/gridplus-sdk/pull/353/commits/756bc001e62e4ba9fce75734645727253137ddc2)